src/honey_badger/epoch_state.rs

```diff
@@ -344,11 +336,13 @@ where
     /// Processes a Threshold Decryption step.
     fn process_decryption(&mut self, proposer_id: N, td_step: td::Step<N>) -> Result<Step<C, N>> {
         let mut step = Step::default();
-        let opt_output = step.extend_with(td_step, |share| {
-            MessageContent::DecryptionShare {
-                proposer_id: proposer_id.clone(),
-                share,
-            }.with_epoch(self.epoch)
+        let opt_output = step.extend_with(td_step, |msg_share| {
+            match msg_share {
+                td::Message::Share(share) => {
+                    MessageContent::DecryptionShare{proposer_id: proposer_id.clone(),
+                                                    share: share}.with_epoch(self.epoch)},
+                _ => panic!(),
```

Comment:
I don't think this panic is the right thing to do, but my rust-fu and hbbt-fu aren't strong enough to know the proper way to handle this. Is there something like unwrap for enums?